nil/common/logging/logger.go

```diff
@@ -55,19 +72,100 @@ func isSystemd() bool {
 	return os.Getenv("INVOCATION_ID") != ""
 }
 
-func NewLogger(component string) zerolog.Logger {
-	var logger zerolog.Logger
+func NewLogger(component string) Logger {
+	var zlogger zerolog.Logger
 	if isSystemd() {
-		logger = newJournalDLogger()
+		zlogger = newJournalDLogger()
+	} else {
+		zlogger = newConsoleLogger()
+	}
+
+	var level zerolog.Level
+	if l, ok := loggerLevelsMap[component]; ok {
+		level = l
 	} else {
-		logger = newConsoleLogger()
+		level = zerolog.GlobalLevel()
 	}
 
-	return logger.With().
+	zlogger = zlogger.With().
 		Str(FieldComponent, component).
 		Caller().
 		Timestamp().
-		Logger()
+		Logger().
+		Level(level)
+
+	lock.Lock()
+	defer lock.Unlock()
+
+	loggersMap[component] = &LoggerRef{Logger: &zlogger}
```

Comment:
I guess you can overwrite existing logger. Could you add an assertion to check that?